server/memstore.go

```diff
@@ -35,6 +35,12 @@ type memStore struct {
 	receivedAny bool
 }
 
+type memStoreMsg struct {
+	*StoreMsg
+	compression StoreCompression // The compression algorithm used
```

Comment:
These can't change per message can they? Rather not introduce additional memory usage per message unless needed.

Reply:
It's possible to change the compression mode of the stream config after creation, hence why we need to know what the compression was set to at the time the message was published. You can't set it on a per-message level otherwise.

Reply:
Should we disallow on memory streams? If they change it what happens?

Reply:
I think we remove, just one compression choice for now.